### ISSUE TITLE
Iran F-4E not pulling custom payload

### ISSUE THREAD

**Issue by Nosajthedevil** [dcs-liberation/dcs_liberation#848]
Before filing, please search the issue tracker to see if the issue has already been reported.

**Describe the bug**
F-4E custom payload ignored for all mission types for Iran. Have not attempted for other factions.

**To Reproduce**
1. Modify F-4E Custom Payload - specifically to give F-4Es AIM-7Es and AIM-9P5s
2. Check Mission Editor
3. Equipped with stock loadout including 7Fs and 9Ms

**Expected behavior**
Expected F-4Es to use AIM-7Es which I'm assuming were what we sold to Iran.

**Additional information**

We will usually need more information for debugging. Include as much of the following as you are able:

- DCS Liberation save file (the `.liberation` file you save from the DCS Liberation window). By default these are located in your DCS saved games directory (`%USERPROFILE%/Saved Games/DCS`).
- The generated mission file (the `.miz` file that you load in DCS to play the turn). By default these are located in your missions directory (`%USERPROFILE%/Saved Games/DCS/Missions`).
- A tacview track file, especially when demonstrating an issue with AI behavior. By default these are locaed in your Tacview tracks directory (`%USERPROFILE%/Documents/Tacview`).

**Version information (please complete the following information):**
 - DCS Liberation 2.4

**Additional context**

**Comment by Nosajthedevil** [comment]
Meant that they were equipped for CAP with 7MHs and 7Ms. All other mission types using 7mh as well

**Comment by DanAlbert** [comment]
I think @Malakhit is the expert on loadouts.

**Comment by Nosajthedevil** [comment]
I have no idea what I'm doing - but in the customized loadouts all non-cap loadouts have been assigned to ["tasks"] = { [1] = 11, which I am reading means CAP. This might or might not be messing something up. . Will edit and try to change later this afternoon or this evening.

**Comment by Nosajthedevil** [comment]
Never mind. Realized that a lot of other payloads are set up that way. I'm assuming payloads are pulled in by name then and not ID number.

**Comment by Malakhit** [comment]
By "modify custom payload" - do you mean that you're tweaking `F-4E.lua`, or do you mean you're modifying the flight loadout in the Liberation UI for that flight specifically?

If you mean the former, would you be able to attach your tweaked `.lua` file, please?

**Comment by Nosajthedevil** [comment]
that's correct, I was tweaking it. Also, I don't know why but it appears that I was correct, when I changed the task ID numbers to match the correct tasks, everything started loading in correctly. Maybe I also fixed something I broke in the first place without realizing it. Anyway, I attached the working version and a nonworking versions. Note I also changed the AIM-9Ls to 9Ps for CAP somewhere in the process. Maybe that did it?
[F-4E.working.txt](https://github.com/Khopa/dcs_liberation/files/5937694/F-4E.working.txt)
[F-4E.notworking.txt](https://github.com/Khopa/dcs_liberation/files/5937695/F-4E.notworking.txt)
